src/resources/users/users.service.ts

```diff
@@ -1,12 +1,144 @@
 /* eslint-disable prettier/prettier */
-import {
-	BadRequestException,
-	Inject,
-	Injectable,
-	InternalServerErrorException,
-} from '@nestjs/common';
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { eq } from 'drizzle-orm';
+import { DrizzleService } from 'src/drizzle/drizzle.service';
+import userContactInfo from 'src/drizzle/schema/user_contact_info';
+import users from 'src/drizzle/schema/users';
+import { AddFcmTokenDto } from './dto/add-fcm-token.dto';
+import { CreateUserDto } from './dto/create-user.dto';
+import { UpdateUserDto } from './dto/update-user.dto';
+import { UserContactInfoDto } from './dto/userContactInfo.dto';
 
 const DEFAULT_CATEGORY_ID = 'default';
 const DEFAULT_LABEL_ID = 'default';
 @Injectable()
-export class UsersService {}
+export class UsersService {
+	constructor(private readonly drizzleService: DrizzleService) {}
+
+	//createUser
+	async createUser(userData: CreateUserDto) {
+		const createdUser = await this.drizzleService.db
+			.insert(users)
+			.values(userData)
+			.returning();
+		return createdUser;
+	}
+
+	//findAll
+	async findAll(page: number, limit: number) {
+		//get total of Pages
+		const totalUsers = await this.drizzleService.db.select().from(users);
+		const totalCount = totalUsers.length;
```

Comment:
Don't Fetch All Rows just to get the count !!!!!
```ts
this.drizzleService.db.select({ count: sql<number>`count(${users.id})` }).from(users)
```

Create a function `userExists(id)` and use it in methods.